src/transport/raw/MessageHeader.h

```diff
@@ -72,33 +72,39 @@ enum class ExFlagValues : uint8_t
     kExchangeFlag_VendorIdPresent = 0x10,
 };
 
-enum class FlagValues : uint16_t
+enum class FlagValues : uint32_t
 {
     /// Header flag specifying that a destination node id is included in the header.
-    kDestinationNodeIdPresent = 0x0100,
+    kDestinationNodeIdPresent = 0x00000100,
 
     /// Header flag specifying that a source node id is included in the header.
-    kSourceNodeIdPresent = 0x0200,
+    kSourceNodeIdPresent = 0x00000200,
 
     /// Header flag specifying that it is a control message for secure session.
-    kSecureSessionControlMessage = 0x0800,
+    kSecureSessionControlMessage = 0x00000800,
 
     /// Header flag specifying that it is a encrypted message.
-    kSecure = 0x0001,
+    kSecure = 0x00000001,
 
+    // Header flag indicates that the peer's group key message counter is not synchronized.
+    kPeerGroupMsgIdNotSynchronized = 0x00010000,
+
+    // Header flag indicates that the message buffer should not be freed after sending.
+    kRetainedBuffer = 0x00020000,
 };
 
-using Flags   = BitFlags<uint16_t, FlagValues>;
+using Flags   = BitFlags<uint32_t, FlagValues>;
```

Comment:
Couldn't the internal flags be its own separate field?  
Put another way, why combine these two fields in the implementation?
```
uint16_t mInternalFlags;
unit16_t mFlags;
```

Reply:
Acked